### PR TITLE
Docs for listing jobs and programs

### LINKED ISSUE #3302: Add list methods for Engine programs and jobs
Use case: retrieving historical data from Quantum Engine.

Solution: support list methods for programs and jobs.

Priority: P2

### PR DESCRIPTION
**Pull request by balopat**
Adds documentation for listing jobs and programs.

Fixes #3302.

### CODE REVIEW

File: docs/google/engine.md [hunk 1]
```diff
@@ -273,4 +273,41 @@ historical_results = historical_job.results()
...
+engine = cirq.google.Engine(project_id='YOUR_PROJECT_ID')
+
+# List all the jobs on the project since 2020/09/20 that succeeded:
+jobs = engine.list_jobs(created_after=datetime.date(2020,9,20),
+                        execution_states=[enums.ExecutionStatus.State.SUCCESS])
+for j in jobs:
+   print(j.job_id, j.status(), j.create_time())
+``` 
+
+Alternatively, you can list programs by creation time and labels, and then list
```
Comment (wcourtney) [inline docs/google/engine.md:294]: Nit: this is a little difficult to parse. Is it worth explaining _how_ they retrieve the program, or could you just say something like "With an existing EngineProgram object, you can list any jobs that were run using that program."?

File: docs/google/engine.md [hunk 2]
```diff
@@ -273,4 +273,41 @@ historical_results = historical_job.results()
...
+
+# List all the programs on the project since 2020/09/20 that have 
+# the "variational" label:
+programs = engine.list_programs(
+                        created_after=datetime.date(2020,9,20),
+                        has_labels={"variational":"*"}
+           )
+for p in programs:
+   print(p.program_id, p.create_time())
+   for j in p.list_jobs():
```
Comment (wcourtney) [inline docs/google/engine.md:310]: Is it worth noting that you can provide additional filters EngineProgram.list_jobs()?

**Review by wcourtney — APPROVED** [review]
LGTM; just some minor nits.

File: docs/google/engine.md [hunk 3]
```diff
@@ -273,4 +273,46 @@ historical_results = historical_job.results()
 ```
 
 If you did not save the ids, you can still find them from your
-job using the [Cloud Console](https://console.cloud.google.com/quantum/jobs).
+job using the [Cloud Console](https://console.cloud.google.com/quantum/jobs) or
+by using `cirq.google.Engine.list_jobs()` or `cirq.google.Engine.list_programs()`.
```
Comment (dstrain115) [inline docs/google/engine.md:277]: It might be worth reiterating the difference between jobs or programs here just so people are clear which one to use.

**Review by dstrain115 — APPROVED** [review]

File: docs/google/engine.md (hunk 3 above)
Comment (balopat) [reply docs/google/engine.md]: Sounds good - I broke out the two sections separately and added a bit more explanation.

**Pull request merged**